app/controllers/admin/discussions_controller.rb

```diff
@@ -0,0 +1,69 @@
+# Admin controller for discussions and comments
+class Admin::DiscussionsController < AdminController
```

Comment:
In Rails, when you see a controller that is not using the conventional RESTful controller actions (`index`, `new`, `create`, etc.), its **sometimes** (perhaps not here) a sign to introduce a controller that would fit the conventional actions.

Here for example, we've got actions `hide`, `unhide`, `lock`, `unlock`, and `*_comment` variations of those too.

Consider if this is a sign to use or introduce `controller#actions` such as:

For discussions:

- `Admin::DiscussionsController#update` handles all hiding/unhiding, locking/unlocking
OR
- `Admin::DiscussionLocksController` and `Admin::DiscussionHidersController`:
    - `DiscussionLocksController#create` -> locks discussion
    - `DiscussionLocksController#destroy` -> unlocks discussion
    - `DiscussionHidersController#create` -> hides discussion
    - `DiscussionHidersController#destroy` -> unhides discussion

For comments:
- `Admin::CommentsController`
    - `#update` handles all hiding/unhiding, locking/unlocking
    - `#destroy` handles destroying
OR
- `Admin::CommentLocksController`, `Admin::CommentHidersController`, and `Admin::CommentsController#destroy`.

I'm not convinced these are appropriate here, but something to consider.

By going with the RESTful routes convention, the controllers tend to have more consistency, and other  advantages often appear throughout the app's development. In particular, it can lead you to a model you didn't know was needed that corresponds to the controller (e.g. `DiscussionLock`, `CommentHider`, etc. - prob. not appropriate here admittedly).